transactions/src/main/java/org/jboss/as/txn/service/JTAEnvironmentBeanService.java

```diff
@@ -56,7 +57,9 @@ public void start(StartContext context) throws StartException {
         // recovery nodes
         jtaEnvironmentBean.setXaRecoveryNodes(Collections.singletonList(nodeIdentifier));
         // setup the XA orphan filters
-        jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(Arrays.asList(JTATransactionLogXAResourceOrphanFilter.class.getName(), JTANodeNameXAResourceOrphanFilter.class.getName(), SubordinateJTAXAResourceOrphanFilter.class.getName()));
+        if (System.getProperty("JTAEnvironmentBean.xaResourceOrphanFilterClassNames") == null) {
```

Comment:
I count 2 lines with my change and 3 with yours:

    if (jtaEnvironmentBean.getXaResourceOrphanFilterClassNames().size() == 0)
        jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(etc)

not that I am arguing that line counting is a good metric for reviews.

As to allowing the user to configure an empty list I would say that using system properties is a back door and the proper way to include this configuration is in the subsystem config and until such an option exists the subsystem should be allowed to apply its own default if the list is empty.